include/unifex/receiver_concepts.hpp

```diff
@@ -195,6 +198,41 @@ constexpr bool is_receiver_cpo_v = is_one_of_v<
 template <typename T>
 using is_receiver_cpo = std::bool_constant<is_receiver_cpo_v<T>>;
 
+template <typename R, typename E>
+UNIFEX_CONCEPT_FRAGMENT(
+  _receiver,
+    requires(remove_cvref_t<R>&& r, E&& e) //
+    (
+      noexcept(set_done(std::move(r))),
```

Comment:
How does this ensure the `noexcept()` expression evaluates to `true` and not just that this is a valid expression?

Reply:
This is a little preprocessor-based DSL. Within a `requires` clause in a `UNIFEX_CONCEPT_FRAGMENT`, `noexcept(expr)` has the same effect as `{ expr } noexcept;` in C++20. (Except for the fact that gcc has a bug in the ABI that prevents this from working in concepts emulation mode. In that case, it only checks that `expr` is well-formed.)